crates/ark-contracts/arkchain/src/orderbook.cairo

```diff
@@ -254,8 +259,46 @@ mod orderbook {
             self.brokers.write(broker_id, 1);
         }
 
+        fn whitelist_creator_broker(ref self: ContractState, broker_id: felt252) {
+            /// todo: change this error to a more specific one
+            // TODO: check components with OZ when ready for ownable.
+            assert(
+                self.admin.read() == starknet::get_caller_address(),
+                orderbook_errors::BROKER_UNREGISTERED
+            );
+
+            if self.creator_brokers.read(broker_id) == 1 {
+                self.creator_brokers.write(broker_id, 0);
+            } else {
+                self.creator_brokers.write(broker_id, 1);
+            }
+        }
+
+        fn whitelist_fulfiller_broker(ref self: ContractState, broker_id: felt252) {
+            /// todo: change this error to a more specific one
+            // TODO: check components with OZ when ready for ownable.
+
+            assert(
+                self.admin.read() == starknet::get_caller_address(),
+                orderbook_errors::BROKER_UNREGISTERED
+            );
+
+            if self.fulfiller_brokers.read(broker_id) == 1 {
+                self.fulfiller_brokers.write(broker_id, 0);
+            } else {
+                self.fulfiller_brokers.write(broker_id, 1);
+            }
+        }
+
+
```

Comment:
I think it should be only one function for both, cause we need to always be able to add new Brokers, then we save the type of broker in each case (create order or fulfill order)